### DISCUSSION TITLE
Feature request to add orientation option and screen size customization

### DISCUSSION THREAD

**Discussion by adriansticoid** [rafaelvcaetano/melonDS-android#1102] [Ideas]
Hi. melonDS at its current state is very good. But I'm missing out on forcing an orientation. Currently when a game is launched, it defaults to portrait orientation. It would be great if we can force it to landscape right away. Also, I think it would be great to add screen size customization. Not a fan of large and small screen layouts.

**Comment by rafaelvcaetano** [comment]
You can already do all that. You just need to create a new layout in Settings > Input > Layouts.
You can adjust the size of the screen as you like for both orientations and force the orientation to whatever you want in the layout's properties.

**Reply by adriansticoid** [reply]
Thank you so much. Although I must say, the location of that setting could be improved. Layout setting does not fit inside the Input setting. Got the layout I wanted now, and I think it would be best to add the option to set the size through the keyboard rather than just a slider. It's faster and easier.

I have two more feature request. First would be the ability to recenter the screen when adjusting the layout. Pizzaboy emulators have this feature. The screen recenters itself when you drag it around. Second would be auto-hiding of the soft buttons for X seconds when input is not detected, and hiding when an external controller is detected.

Thank you so much Rafael for responding and listening to users like me. Really appreciate it.

**Comment by Pokepal7** [comment]
> You can already do all that. You just need to create a new layout in Settings > Input > Layouts. You can adjust the size of the screen as you like for both orientations and force the orientation to whatever you want in the layout's properties.

Not exactly true, the max size is 1080 only, in my case i want to make main screen more bigger than touch screen and full screen.
I want the main screen to ocupate more screen of my phone than touch screen, drastic can do that.
